crates/subspace-networking/src/protocols/autonat_wrapper.rs

```diff
@@ -36,14 +36,13 @@ impl Behaviour {
     }
 
     fn address_corresponds_to_listening_addresses(&self, addr: &Multiaddr) -> bool {
-        let candidate_protocol = addr
-            .iter()
-            .find_map(|protocol| match protocol {
-                udp @ Protocol::Udp(_) => Some(udp),
-                tcp @ Protocol::Tcp(_) => Some(tcp),
-                _ => None,
-            })
-            .expect("Either TCP or UDP protocol should be enabled.");
+        let Some(candidate_protocol) = addr.iter().find_map(|protocol| match protocol {
+            udp @ Protocol::Udp(_) => Some(udp),
+            tcp @ Protocol::Tcp(_) => Some(tcp),
+            _ => None,
```

Comment:
Shouldn't we capture QuicV1 or Quic here now that its properly translated ?

Reply:
Not really, quic is the only UDP protocol we use, so the only thing we care about here are TCP/UDP ports to avoid mismatches. If we were pedantic about it we could check the rest, but dealing with a single protocol rather than a list makes code much shorter.